### ISSUE TITLE
Authorisation request does not seem to be using client certificate

### ISSUE THREAD

**Issue by tobscher** [docker/docker#13584]
## Description of problem

It seems that when I try to push to a private registry which has been configured to use token authentication that the request made to obtain a token is not using a client certificate. I have the following setup:

* Registry: docker-registry.mydomain.biz
* Authorisation: docker-registry-auth.mydomain.biz

Client certs have been put in `/etc/docker/certs.d`:
```
$ tree /etc/docker/certs.d/
/etc/docker/certs.d/
├── docker-registry-auth.mydomain.biz
│   ├── client.cert
│   ├── client.key
│   └── docker-registry-auth.mydomain.biz.crt
└── docker-registry.mydomain.biz
    ├── client.cert
    ├── client.key
    └── docker-registry.mydomain.biz.crt

2 directories, 6 files
```

In my registry config I have `auth.token.realm` set to docker-registry-auth.mydomain.biz/token.

I can see in the logs that the request made to the registry is using the client certificate. The request made to my authorisation service result in the following:

```
2015/05/29 09:29:57 http: TLS handshake error from xxxxxxx: tls: client didn't provide a certificate
```

Here is a snippet from the authorisation server code:

```go
package main

import (
	"flag"
	"log"
	"net/http"
	"time"
	"crypto/tls"

	"github.com/tobscher/docker-registry-authentication/server"
)

var (
	binding  = flag.String("binding", ":5000", "http server binding")
	certFile = flag.String("cert", "cert.pem", "A PEM eoncoded certificate file.")
	keyFile  = flag.String("key", "key.pem", "A PEM encoded private key file.")
)

func main() {
	flag.Parse()

	context := &server.Context{}
	http.Handle("/token", server.TokenHandler(context))

	server := &http.Server{
		Addr: *binding,
		TLSConfig: &tls.Config{
			ClientAuth: tls.RequireAndVerifyClientCert,
		},
	}

	log.Printf("Binding: %v\n", *binding)
	if err := server.ListenAndServeTLS(*certFile, *keyFile); err != nil {
		log.Fatal(err)
	}
}
```

docker version:
```
Client version: 1.6.0
Client API version: 1.18
Go version (client): go1.3.3
Git commit (client): 4749651/1.6.0
OS/Arch (client): linux/amd64
Server version: 1.6.0
Server API version: 1.18
Go version (server): go1.3.3
Git commit (server): 4749651/1.6.0
OS/Arch (server): linux/amd64
```

docker info:
```
Containers: 0
Images: 3
Storage Driver: devicemapper
 Pool Name: docker-202:1-263472-pool
 Pool Blocksize: 65.54 kB
 Backing Filesystem: extfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 310.4 MB
 Data Space Total: 107.4 GB
 Data Space Available: 6.825 GB
 Metadata Space Used: 806.9 kB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.147 GB
 Udev Sync Supported: true
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.89-RHEL6 (2014-09-01)
Execution Driver: native-0.2
Kernel Version: 3.14.35-28.38.amzn1.x86_64
Operating System: Amazon Linux AMI 2015.03
CPUs: 1
Total Memory: 3.68 GiB
Name: ip-10-1-0-12
ID: LPJU:3T44:NZRF:PYPS:L2LE:2EKS:DNYW:TSYA:3OEJ:JBZL:T3OR:PXGO
Debug mode (server): true
Debug mode (client): false
Fds: 14
Goroutines: 15
System Time: Fri May 29 09:31:50 UTC 2015
EventsListeners: 0
Init SHA1: e873b920c3551f59f6eac8f69eb14d11e770d445
Init Path: /usr/libexec/docker/dockerinit
Docker Root Dir: /var/lib/docker
```

uname -a:
```
Linux ip-10-1-0-12 3.14.35-28.38.amzn1.x86_64 #1 SMP Wed Mar 11 22:50:37 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```

Environment details: AWS

## Steps to replicate

* Follow the guides at https://docs.docker.com/registry/deploying/
* Configure token authentication
* Try to push a repository to `docker-registry.mydomain.biz/username/repo`

## Actual results

* Authorisation request does not include client certificate

## Expected results

* Authorisation request uses client certificate from `/etc/docker/certs.d/docker-registry-auth.mydomain.biz`

/cc @rosstimson

**Comment by excpt-one** [comment]
Have the same issue.
uname -a:
```
Linux excpt-pc 3.16.0-34-generic #45~14.04.1-Ubuntu SMP Tue Mar 24 11:14:29 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```
docker version:
```
Client version: 1.6.2
Client API version: 1.18
Go version (client): go1.4.2
Git commit (client): 7c8fca2
OS/Arch (client): linux/amd64
Server version: 1.6.2
Server API version: 1.18
Go version (server): go1.4.2
Git commit (server): 7c8fca2
OS/Arch (server): linux/amd64
```
nginx response:
```
400 Bad Request
No required SSL certificate was sent
```

**Comment by tobscher** [comment]
I retested it with the current version from master which now seems to be sending a client certificate when making an authorisation request:

docker version:
```
Client version: 1.7.0-dev
Client API version: 1.20
Go version (client): go1.4.2
Git commit (client): 521926f
OS/Arch (client): linux/amd64
Server version: 1.7.0-dev
Server API version: 1.20
Go version (server): go1.4.2
Git commit (server): 521926f
OS/Arch (server): linux/amd64
```

**Comment by tobscher** [comment]
I am going to close this issue as it seems to be resolved in docker 1.7
